sdk/maps/Azure.Maps.Search/tests/ReverseSearchAddressTests.cs

```diff
@@ -78,10 +79,10 @@ public async Task CanPoolReverseSearchAddressBatch()
                 new ReverseSearchAddressQuery(new ReverseSearchOptions { coordinates = new GeoPosition(121.0, 24.0), Language = "en" }),
                 new ReverseSearchAddressQuery(new ReverseSearchOptions { coordinates = new GeoPosition(-122.333345, 47.606038) }),
             });
-
+            Thread.Sleep(500);
```

Comment:
This should be `Task.Delay` since you're running in an asynchronous context.   That said, if we can make the test approach more deterministic rather than resorting to arbitrary delays, that'd be a much better idea.

Why does it matter there is a delay between the LRO call and waiting for completion?   I'd expect customer code to do exactly what this test had been doing.

Reply:
We always have the unit test error in MacOS environment for LRO. Like this one: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1832647&view=logs&j=e1233b80-fe26-54b1-8da9-2330cdd5d8f1&t=e42962a0-eaf5-52a0-a3b4-15bb9b250940&l=706
It's flacky, but I don't know why. If we sleep for few milliseconds, we should be able the fix this issue. Do you have any thoughts why LRO only fails in MacOS environment? Is there any way we can avoid or fix this issue if we don't add `Task.Delay`?

Reply:
There's no known issue of which I'm aware.  Without context of some sort, there's nothing that I can even offer speculation for.  Do you have a stacktrace or other information around what you believe is different specific to macOS behavior?

Reply:
I guess it's performance/timing issue. If we send LRO request and ask for result immediately (like what we have in unit test), and server side hasn't finish processing the request yet, it can result in 404 request not found. I've added `Task.Dealy`. Should avoid the flacky test result now.